contracts/plugins/trading/DutchTrade.sol

```diff
@@ -183,6 +199,50 @@ contract DutchTrade is ITrade {
         bestPrice = _bestPrice; // gas-saver
     }
 
+    /// Bid with callback for the auction lot at the current price;
+    ///  Sold funds are sent back to the callee, callee.dutchTradeCallback(...) is invoked
+    ///  balance of buy token must increase by bidAmount(current block) after callback
+    ///  Trade is settled atomically via a callback
+    ///
+    /// @param data {bytes} The data to pass to the callback
+    /// @dev Caller must implement IDutchTradeCallee
+    /// @return amountIn {qBuyTok} The quantity of tokens the bidder paid
+    function bidWithCallback(bytes calldata data) external returns (uint256 amountIn) {
+        require(bidder == address(0), "bid already received");
+        // {buyTok/sellTok}
+        uint192 price = _price(block.number); // enforces auction ongoing
+
+        // {qBuyTok}
+        amountIn = _bidAmount(price);
+
+        // Mark bidder
+        bidder = msg.sender;
```

Comment:
this section is not parallel between the two functions. I think all the assignments should be grouped as one section and then the `safeTransferFrom()` on line 261 should stand on its own.

Reply:
Reordered the implementations to correctly reflect each other.